src/main/drivers/timer_hal.c

```diff
@@ -801,75 +816,96 @@ _TIM_IRQ_HANDLER(TIM8_BRK_TIM12_IRQHandler, 12);
 _TIM_IRQ_HANDLER(TIM1_BRK_TIM15_IRQHandler, 15);
 #endif
 #if defined(STM32F303xC) && ((USED_TIMERS & (TIM_N(1)|TIM_N(16))) == (TIM_N(16)))
-_TIM_IRQ_HANDLER(TIM1_UP_TIM16_IRQHandler, 16);    // only timer16 is used, not timer1
+_TIM_IRQ_HANDLER(TIM1_UP_TIM16_IRQHandler, 16);         // only timer16 is used, not timer1
 #endif
 #if USED_TIMERS & TIM_N(17)
 _TIM_IRQ_HANDLER(TIM1_TRG_COM_TIM17_IRQHandler, 17);
 #endif
 
-void timerInit(void)
+#ifdef USE_TIMER_MGMT
+TIM_TypeDef* getTimByTag(const timerTag_t tag)
 {
-    memset(timerConfig, 0, sizeof (timerConfig));
-
+#define _CASE(n) case n : return TIM ## n
+    switch (TIMER_TAG_NO(tag)) {
 #if USED_TIMERS & TIM_N(1)
-    __HAL_RCC_TIM1_CLK_ENABLE();
+    _CASE(1);
 #endif
 #if USED_TIMERS & TIM_N(2)
-    __HAL_RCC_TIM2_CLK_ENABLE();
+    _CASE(2);
 #endif
 #if USED_TIMERS & TIM_N(3)
-    __HAL_RCC_TIM3_CLK_ENABLE();
+    _CASE(3);
 #endif
 #if USED_TIMERS & TIM_N(4)
-    __HAL_RCC_TIM4_CLK_ENABLE();
+    _CASE(4);
 #endif
 #if USED_TIMERS & TIM_N(5)
-    __HAL_RCC_TIM5_CLK_ENABLE();
+    _CASE(5);
 #endif
 #if USED_TIMERS & TIM_N(6)
-    __HAL_RCC_TIM6_CLK_ENABLE();
+    _CASE(6);
 #endif
 #if USED_TIMERS & TIM_N(7)
-    __HAL_RCC_TIM7_CLK_ENABLE();
+    _CASE(7);
 #endif
 #if USED_TIMERS & TIM_N(8)
-    __HAL_RCC_TIM8_CLK_ENABLE();
+    _CASE(8);
 #endif
 #if USED_TIMERS & TIM_N(9)
-    __HAL_RCC_TIM9_CLK_ENABLE();
+    _CASE(9);
 #endif
 #if USED_TIMERS & TIM_N(10)
-    __HAL_RCC_TIM10_CLK_ENABLE();
+    _CASE(10);
 #endif
 #if USED_TIMERS & TIM_N(11)
-    __HAL_RCC_TIM11_CLK_ENABLE();
+    _CASE(11);
 #endif
 #if USED_TIMERS & TIM_N(12)
-    __HAL_RCC_TIM12_CLK_ENABLE();
+    _CASE(12);
 #endif
 #if USED_TIMERS & TIM_N(13)
-    __HAL_RCC_TIM13_CLK_ENABLE();
+    _CASE(13);
 #endif
 #if USED_TIMERS & TIM_N(14)
-    __HAL_RCC_TIM14_CLK_ENABLE();
+    _CASE(14);
 #endif
 #if USED_TIMERS & TIM_N(15)
-    __HAL_RCC_TIM15_CLK_ENABLE();
+    _CASE(15);
 #endif
 #if USED_TIMERS & TIM_N(16)
-    __HAL_RCC_TIM16_CLK_ENABLE();
+    _CASE(16);
 #endif
 #if USED_TIMERS & TIM_N(17)
-    __HAL_RCC_TIM17_CLK_ENABLE();
+    _CASE(17);
+#endif
+    default: return NULL;
+    }
+#undef _CASE
+}
 #endif
 
-    /* enable the timer peripherals */
-    for (int i = 0; i < USABLE_TIMER_CHANNEL_COUNT; i++) {
-        RCC_ClockCmd(timerRCC(timerHardware[i].tim), ENABLE);
+void timerInit(void)
+{
+    memset(timerConfig, 0, sizeof(timerConfig));
+
+#ifdef USE_TIMER_MGMT
+    for (unsigned i = 0; i < TIMER_CHANNEL_COUNT; i++) {
+        timerHardware_t *th = &timerHardware[i];
+        th->tim = getTimByTag(timerTags[i]);
+        th->tag = timerChannelConfig(i)->ioTag;
+        th->channel = (TIMER_TAG_CHANNEL_INDEX(timerTags[i]) << 2);
+        th->output = timerChannelConfig(i)->inverted ? TIMER_OUTPUT_N_CHANNEL : TIMER_OUTPUT_NONE;
+        th->alternateFunction = timerAlternateFunction(timerTags[i], timerChannelConfig(i)->ioTag);
```

Comment:
(this may be inferred when `alternateFunction` is needed)

Reply:
Maybe, but needed currently for backward compatibility. That clean up can be a separate PR